### ISSUE TITLE
add script-based unlocking

### ISSUE THREAD

**Issue by GlenDC** [rivine/rivine#174]
blocked by #173 and #167.

Given the structure designed and developed for #173, we'll implement and register an `InputUnlocker` for the `UnlockTypeScript` (`0x02`).

+ unlockCondition = conditionScript (what has to be done to spend coins);
+ unlockFulfillment = fulfillmentScript (push stuff onto the stack only, or in other words, prepare the stack with the needed input);

Validation:
1. start with an empty stack
2. run fulfillmentScript 
3. run conditionScript

Validation and execution of those scripts, will be done using the Rivine Scripting Engine module, which is also what implements the needed components (e.g. the stack).

All steps have to exit without errors, and conditionScript has to return `True`.

further there are some additional rules:
+ op codes have to be valid
+ param instructions within an opcode have to be valid
+ fulfillment scripts can only push data to the stack, no other op codes can be used there
+ ...

**Comment by GlenDC** [comment]
No longer needed for now, see #184.
